bodo/spawn/worker.py

```diff
@@ -440,13 +439,19 @@ def exec_func_handler(
         return
 
     is_distributed = False
-    if func is not None and len(func.signatures) > 0:
+    if is_dispatcher and func is not None and len(func.signatures) > 0:
         # There should only be one signature compiled for the input function
         sig = func.signatures[0]
         assert sig in func.overloads
 
         # Extract return value distribution from metadata
         is_distributed = func.overloads[sig].metadata["is_return_distributed"]
+
+    # TODO: handle other types
```

Comment:
Could you add a bit more to the comment. Clarifying that it must be a compiler-representable type?

Reply:
Added a comment. The output doesn't need to be JITable since we can just pickle (as we do with output scalars). The challenge is that scalars may not be replicated in the non-JIT cases like map_partitions, so we have to define
the semantics (e.g. gather all values across ranks in a list).

Reply:
What does `bodo.typeof` return if its not jittable? Or a type not known in the Numba type registry

Reply:
I agree, this needs a try/except around it probably. I was focused on the larger design issue.